core/src/main/java/org/python/core/PyLong.java

```diff
@@ -0,0 +1,1415 @@
+                return r;
+            else if (PyLong.TYPE.check(r)) {
+                // 'result' not of exact type int but is a subclass
+                Abstract.returnDeprecation("__index__", "int", r);
+                return r;
+            } else
+                throw Abstract.returnTypeError("__index__", "int", r);
+        } catch (EmptyException e) {}
+
+        // We're here because op_index was empty. Try op_int.
+        if (Slot.op_int.isDefinedFor(ops)) {
+            Object r = fromIntOf(integral);
+            // ... but grumble about it.
+            // Warnings.format(DeprecationWarning.TYPE, 1,
+            // "an integer is required (got type %.200s). "
+            // + "Implicit conversion to integers "
+            // + "using __int__ is deprecated, and may be "
+            // + "removed in a future version of Python.",
+            // ops.type(integral).name);
+            return r;
+        } else
+            throw Abstract.requiredTypeError("an integer", integral);
+    }
+
+    /**
+     * Convert a sequence of Unicode digits in the string u to a Python
+     * integer value.
+     *
+     * @param u string to convert
+     * @param base in which to interpret it
+     * @return converted value
+     * @throws ValueError if {@code u} is an invalid literal
+     * @throws TypeError if {@code u} is not a Python {@code str}
+     */
+    // Compare CPython longobject.c :: PyLong_FromUnicodeObject
+    static BigInteger fromUnicode(Object u, int base) throws ValueError, TypeError {
+        try {
+            // XXX maybe check 2<=base<=36 even if Number.asLong does?
+            // XXX Should allow for only string types, but for now ...
+            String value = u.toString();
+            // String value = PyUnicode.asString(u);
+            return new BigInteger(value, base);
+        } catch (NumberFormatException e) {
+            throw new ValueError("invalid literal for int() with base %d: %.200s", base, u);
+        }
+    }
+
+    /**
+     * Return a Python {@code int} from a Python {@code int} or
+     * subclass. If the value has exactly Python type {@code int} return
+     * it, otherwise construct a new instance of exactly {@code int}
+     * type.
+     *
+     * @param value to represent
+     * @return the same value as exactly {@code int}
+     * @throws TypeError if not a Python {@code int} or sub-class
+     */
+    static Object from(Object value) throws TypeError {
+        Operations ops = Operations.of(value);
+        if (ops.isIntExact())
+            return value;
+        else if (value instanceof PyLong)
+            return ((PyLong)value).value;
+        else
+            throw Abstract.requiredTypeError("an integer", value);
+    }
+
+
+    // Methods --------------------------------------------------------
+    // Expose to Python when mechanisms are available
+
+    /*
+
+    @ExposedGet(name = "real", doc = BuiltinDocs.long_real_doc)
+     */
```

Comment:
It occurred to me that dual docs for Java and Python docstrings could be handled using Javadoc's [@value](https://docs.oracle.com/en/java/javase/16/docs/specs/javadoc/doc-comment-spec.html#value) tag.
E.g. in this case the javadoc would be
```
/**
 * {@value BuiltinDocs#long_real_doc}
 */
```

Reply:
That's a clever idea. The actual doc string in code is not quite what you see in help (has some mark-up) but is close enough to be readable by maintainers.

These annotations from Jython 2 are preserved in this odd form to make it easier to generate a diff that aligns with the old code. (`kdiff3` needed these clues.)

I have equivalent annotations in the experimental code off-repo, including one for docstrings, but I'm not sure how it should work. Something similar to Jython 2 would be good, with the objective that strings don't load until used. I couldn't tell if Jython 2 achieves that. It's not my top thing to work on.